migration/i8.2/oracle/202106291050_IDEMPIERE-4845.sql

```diff
@@ -0,0 +1,17 @@
+SET SQLBLANKLINES ON
+SET DEFINE OFF
+
+-- IDEMPIERE-4845
+-- 23.06.2021, 10:26:41 MESZ
+INSERT INTO AD_Message (MsgType,MsgText,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Message_ID,Value,EntityType,AD_Message_UU) VALUES ('E','No product with specified attribute set instance is stored: {0}',0,0,'Y',TO_DATE('2021-06-23 10:26:35','YYYY-MM-DD HH24:MI:SS'),100,TO_DATE('2021-06-23 10:26:35','YYYY-MM-DD HH24:MI:SS'),100,200712,'NoProductWithAsi','D','98925dc0-f2fb-4801-be0e-553904c92377')
+;
+-- AMEND-IDEMPIERE-4845
+-- 28.06.2021, 08:43:38 MESZ
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,Help,AD_Table_ID,ColumnName,DefaultValue,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure,FKConstraintType,IsHtml) VALUES (214520,0,'Auto Generated','Record is Auto Generated by System.',765,'IsAutoGenerated','N',1,'N','N','N','N','N',0,'N',20,0,0,'Y',TO_DATE('2021-06-28 08:43:33','YYYY-MM-DD HH24:MI:SS'),100,TO_DATE('2021-06-28 08:43:33','YYYY-MM-DD HH24:MI:SS'),100,202701,'N','N','D','N','N','N','Y','442e9fd0-9587-4ea3-9f77-2126731561bf','Y',0,'N','N','N','N')
+;
+
+-- 28.06.2021, 08:43:45 MESZ
+ALTER TABLE M_ProductionLineMA ADD IsAutoGenerated CHAR(1) DEFAULT 'N' CHECK (IsAutoGenerated IN ('Y','N'))
```

Comment:
this default is wrong for existing data, should be Y instead.

Reply:
While that is true, there is no mechanical difference when creating attribute set instances manually as opposed to automatically, so setting the default to 'Y' would make it impossible to mark them as manually created at all. The option I see to mark all old entries as automatically generated would be to add
`UPDATE M_ProductionLineMA SET IsAutoGenerated = 'Y';`
Since this script is only run once, that should solve the issue.

Reply:
> UPDATE M_ProductionLineMA SET IsAutoGenerated = 'Y';
>  Since this script is only run once, that should solve the issue.

I think is safer to change the default to 'Y' - is better not to assume the script won't be re-run.

Reply:
You will break existing integration and customization code that writes directly to m_productionlinema.

Reply:
Hi @hengsin - don't know if the comment is for me or for @lHeidbreder 

I have not analyzed the code to see backward compatibility.  So, if you think the default must be 'N' is OK, but in such case I think to be on the safe side that must be done in three steps:

1 - add the column with default null and then:
 UPDATE M_ProductionLineMA SET IsAutoGenerated = 'Y' WHERE IsAutoGenerated IS NULL;
(which is the same as adding the column with default='Y')

2 - and then change the default to 'N' for the new records

My point is, it must not be assumed that people will not run the script again, there must be a where clause to protect against that case.

Reply:
Hi @CarlosRuiz-globalqss ,

My comments above is for @lHeidbreder. I'm the first to raise above that the default should be 'Y' instead of 'N'.

Regards,
Low